### PR TITLE
Adds a new "Highlighted" checkbox in V6.

### PR DESCRIPTION
**Pull request by mthadley**
This adds a new `Highlighted` constructor to the `Theme` type in `Checkbox`. This new theme will be used for some new features on the Curriculum Library.

Here's what it looks like:

highlighted_checkbox

Since `PremiumCheckbox` actually depends on `Checkbox`, I decided to also create a new version of `PremiumCheckbox` that would be compatible with the newest version of `Checkbox`. I'm not sure if that was the right move, so please take a look at that change.

Thanks!

### CODE REVIEW

File: src/Nri/Ui/Checkbox/V6.elm [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+    , disabled : Bool
+    , theme : Theme
+    }
+
+
+{-|
+
+    = Selected --  Checked (rendered with a checkmark)
+    | NotSelected -- Not Checked (rendered blank)
+    | PartiallySelected -- Indeterminate (rendered dash)
```
Comment (avh4) [inline src/Nri/Ui/Checkbox/V6.elm]: 🎨 4-space indent will render as a code block.

suggested:
```
- `Selected`: Checked (rendered with a checkmark)
- `NotSelected`: Not Checked (rendered blank)
- `PartiallySelected`: Indeterminate (rendered dash)
```

File: src/Nri/Ui/Checkbox/V6.elm [hunk 2]
```diff
@@ -369,6 +373,100 @@ checkboxUnchecked =
         |> Icon
 
 
+checkboxCheckedHighlighted : Icon
+checkboxCheckedHighlighted =
```
Comment (avh4) [inline src/Nri/Ui/Checkbox/V6.elm]: ❓ why is the check itself a complete different icon?

File: src/Nri/Ui/Checkbox/V6.elm (hunk 1 above)
Comment (mthadley) [reply src/Nri/Ui/Checkbox/V6.elm]: This was copied from the previous module version. I think they may have actually intended to for this to be rendered as a code block? It's like, sort of Elm code, after all. Either way, I think your suggestion is nicer to read.

File: src/Nri/Ui/Checkbox/V6.elm (hunk 2 above)
Comment (mthadley) [reply src/Nri/Ui/Checkbox/V6.elm]: This was the asset I received from design. We could also style the existing checked icon based on the current theme.

**Comment by mthadley** [comment]
> ❓ This new "theme" has completed different spacing, padding, and layout properties -- this seems like it's meant to be used in a completely different way than the normal checkbox; is that the intent from the design team? Also why does this highlighted one have a different icon from normal checkboxes -- should the normal checkboxes use the new icon?

Oh, I actually forgot to update the new checked icon to match the dimensions of the other icons. It doesn't look as out of place with that change.

As far as the intention of the new icon, we can ping @NoRedInk/design about it. Was it intended that the checkbox use the old "unchecked" style, and the have this new style when "checked"?

**Comment by bendansby** [comment]
My intention was for us to just swap in a new graphic for the checked state so it has a blue border

File: src/Nri/Ui/Checkbox/V6.elm (hunk 2 above)
Comment (mthadley) [reply src/Nri/Ui/Checkbox/V6.elm]: Also, I wonder if the only SVG assets we really need in this module are the lock and the checkmark. The rest of the checkbox could be CSS.

**Comment by bendansby** [comment]
I'm wondering if this needs to be part of checkbox? It seems like the highlighting is more specific to the CL. Reminds me of this situation https://github.com/NoRedInk/noredink-ui/pull/296

**Comment by mthadley** [comment]
> I'm wondering if this needs to be part of checkbox? It seems like the highlighting is more specific to the CL. Reminds me of this situation #296

Hah, yes! Very similar.

If we don't plan on having more checkboxes use this "style" in the future, than we should just implement this only on the CL.

I also originally thought I needed to make changes this module to actually get the styles to match, but it sounds like I can just add that blue border to the SVG and call it a day.

Unless there are any objecting thoughts, I can close this and go that route instead.

**Pull request closed without merging**